application/actions/requests.go

```diff
@@ -186,3 +185,125 @@ func convertRequestInput(ctx context.Context, input api.RequestCreateInput) (mod
 
 	return request, nil
 }
+
+// swagger:operation PUT /requests/{request_id} Requests RequestsUpdate
+//
+// update a request
+//
+// ---
+// parameters:
+//   - name: RequestUpdateInput
+//     in: body
+//     required: true
+//     description: input object
+//     schema:
+//       "$ref": "#/definitions/RequestUpdateInput"
+//
+// responses:
+//   '200':
+//     description: the request
+//     schema:
+//       "$ref": "#/definitions/Request"
+func requestsUpdate(c buffalo.Context) error {
+	var input api.RequestUpdateInput
+	if err := StrictBind(c, &input); err != nil {
+		err = errors.New("unable to unmarshal data into RequestCreateInput, error: " + err.Error())
+		return reportError(c, api.NewAppError(err, api.ErrorInvalidRequestBody, api.CategoryUser))
+	}
+
+	tx := models.Tx(c)
+
+	requestID, err := getUUIDFromParam(c, "request_id")
+	if err != nil {
+		return reportError(c, err)
+	}
+
+	request, err := convertRequestUpdateInput(c, input, requestID.String())
+	if err != nil {
+		return reportError(c, err)
+	}
+
+	if err = request.Update(tx); err != nil {
+		appError := api.NewAppError(err, api.ErrorUpdateRequest, api.CategoryUser)
+		// TODO: check for validation errors
+		if domain.IsOtherThanNoRows(err) {
+			appError.Category = api.CategoryInternal
+		}
+		return reportError(c, appError)
+	}
+
+	output, err := models.ConvertRequest(c, request)
+	if err != nil {
+		return reportError(c, err)
+	}
+
+	return c.Render(200, render.JSON(output))
+}
+
+// convertRequestUpdateInput updates a `Request` from a `RequestUpdateInput`, with the values in
+// the database as default for any property not specified in the input.
+func convertRequestUpdateInput(ctx context.Context, input api.RequestUpdateInput, id string) (models.Request, error) {
+	tx := models.Tx(ctx)
+
+	var request models.Request
+	if err := request.FindByUUID(tx, id); err != nil {
+		appError := api.NewAppError(err, api.ErrorUpdateRequest, api.CategoryNotFound)
+		if domain.IsOtherThanNoRows(err) {
+			appError.Category = api.CategoryInternal
+		}
+		return request, appError
+	}
+
+	if input.PhotoID.Valid {
+		if _, err := request.AttachPhoto(tx, input.PhotoID.UUID.String()); err != nil {
```

Comment:
I guess I didn't see this when reviewing the RequestCreate, but anyway ...

I'm wondering if in the long run, these lines that actually touch the database (here and below) should be moved to the model's update function and this function should only do the transfer of values from the input object to the new request.

If you don't agree, leave it as is, or if you want, you can make it a "TODO".

Reply:
Started to work on it because I think you're right, but it's a significant change